addons/payment/models/payment_token.py

```diff
@@ -83,8 +83,9 @@ def write(self, values):
         """
         if 'active' in values:
             if values['active']:
-                if any(not token.active for token in self):
-                    raise UserError(_("A token cannot be unarchived once it has been archived."))
+                for token in self:
+                    if not(token.payment_method_id.active and not token.provider_id.state == "disabled"):
+                        raise UserError(_("Once you archive the payment method and provider, you can't unarchive the token."))
```

Comment:
Why did you remove any? :thinking:
Also you can move you `not` operator inside to make condition little simpler. No?

Reply:
Done the changes :+1: